### PR TITLE
Adding support for mapstruct

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- support for mapstruct

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/contact/ContactMapper.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @Mapping(target = "customerId", source = "customers.customerId")
+    @Mapping(target = "phoneNumber", source = "phone")
+    Contact toContact(Contacts contacts);
+
+    @Named("toCustomers")
+    static Customers toCustomers(BigDecimal customerId) {
+        return Customers.findById(customerId.toBigInteger());
+    }
+
+}
```
Comment (sdevalapurkar) [inline src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/contact/ContactMapper.java:28]:
```suggestion
}

```

**Review by sdevalapurkar — APPROVED** [review]
thank you! LGTM

File: src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/contact/ContactMapper.java (hunk 1 above)
Comment (akroon3r) [inline src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/contact/ContactMapper.java:28]:
```suggestion

}
```

File: src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/controller/ContactsApiServiceImpl.java [hunk 2]
```diff
@@ -17,20 +16,31 @@
...
 
+    private final ContactsService contactsService;
+    private final ContactMapper contactMapper;
+
+    public ContactsApiServiceImpl(ContactsService contactsService, ContactMapper contactMapper) {
+        this.contactsService = contactsService;
+        this.contactMapper = contactMapper;
+    }
+
+
```
Comment (akroon3r) [inline src/backend/quarkus-management-api/src/main/java/com/nttdata/quarkus/management/api/controller/ContactsApiServiceImpl.java:34]:
```suggestion
```

**Review by akroon3r — COMMENTED** [review]
two smol changes

**Pull request merged**